zerver/tests/test_subs.py

```diff
@@ -535,6 +540,98 @@ def get_unread_stream_data(user: UserProfile) -> list[UnreadStreamInfo]:
+        result = self.common_subscribe_to_streams(user, subscriptions, subdomain="zulip")
+        self.assert_json_success(result)
+        stream = get_stream("new_stream2", realm)
+        self.assertEqual(stream.can_administer_channel_group.id, nobody_system_group.id)
+
+        hamletcharacters_group = NamedUserGroup.objects.get(name="hamletcharacters", realm=realm)
+        subscriptions = [{"name": "new_stream3", "description": "Third new stream"}]
+        result = self.common_subscribe_to_streams(
+            user,
+            subscriptions,
+            {"can_administer_channel_group": orjson.dumps(hamletcharacters_group.id).decode()},
+            allow_fail=True,
+            subdomain="zulip",
+        )
+        self.assert_json_success(result)
+        stream = get_stream("new_stream3", realm)
+        self.assertEqual(stream.can_administer_channel_group.id, hamletcharacters_group.id)
+
+        subscriptions = [{"name": "new_stream4", "description": "Fourth new stream"}]
+        result = self.common_subscribe_to_streams(
+            user,
+            subscriptions,
+            {
+                "can_administer_channel_group": orjson.dumps(
+                    {"direct_members": [user.id], "direct_subgroups": [moderators_system_group.id]}
+                ).decode()
+            },
+            allow_fail=True,
+            subdomain="zulip",
+        )
+        self.assert_json_success(result)
+        stream = get_stream("new_stream4", realm)
+        self.assertEqual(list(stream.can_administer_channel_group.direct_members.all()), [user])
+        self.assertEqual(
+            list(stream.can_administer_channel_group.direct_subgroups.all()),
+            [moderators_system_group],
+        )
+
+        subscriptions = [{"name": "new_stream5", "description": "Fifth new stream"}]
+        internet_group = NamedUserGroup.objects.get(
+            name="role:internet", is_system_group=True, realm=realm
+        )
+        result = self.common_subscribe_to_streams(
+            user,
+            subscriptions,
+            {"can_administer_channel_group": orjson.dumps(internet_group.id).decode()},
+            allow_fail=True,
+            subdomain="zulip",
+        )
+        self.assert_json_error(
+            result,
+            "'can_administer_channel_group' setting cannot be set to 'role:internet' group.",
+        )
+
+        everyone_group = NamedUserGroup.objects.get(
+            name="role:everyone", is_system_group=True, realm=realm
+        )
+        result = self.common_subscribe_to_streams(
+            user,
+            subscriptions,
+            {"can_administer_channel_group": orjson.dumps(everyone_group.id).decode()},
+            allow_fail=True,
+            subdomain="zulip",
+        )
+        self.assert_json_error(
+            result,
+            "'can_administer_channel_group' setting cannot be set to 'role:everyone' group.",
+        )
```

Comment:
I think we can have these checks in a function and that function can be called for each setting. We do the same for group settings. Can probably be done as a follow-up.